subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/AbstractRecompilationSpecProvider.java

```diff
@@ -117,6 +118,13 @@ private String rebuildClauseForChangedNonSourceFile(FileChange fileChange) {
         return String.format("%s '%s' has been %s", "resource", fileChange.getFile().getName(), fileChange.getChangeType().name().toLowerCase(Locale.US));
     }
 
+    protected abstract void processGroovyJavaJointCompilationAdditionalDependencies(
```

Comment:
I don't like this method name. Why would `AbstractRecompilationSpecProvider` need to know about Groovy?

Reply:
Yeah, I had same thoughts, basically this abstract class becomes tightly coupled with sub classes. So I was thinking to have something like "processAdditionalDependencies", but then for readability reasons (so you immediately know this method is there for joint compilation), I went with this name anyway.

Do you have any suggestion?

Reply:
How about `addLanguageSpecificDependencies` or `processLanguageSpecificDependencies? Or maybe using `Compiler` instead of `Language` also makes sense. WDYT?

Reply:
I like it, I will use the option with compiler so: `processCompilerSpecificDependencies`